src/DiagnosticSourceListener/DiagnosticSourceTelemetryModule.cs

```diff
@@ -38,9 +38,16 @@ public void Initialize(TelemetryConfiguration configuration)
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            this.client = new TelemetryClient(configuration);
-            this.client.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("dsl:");
-            this.allDiagnosticListenersSubscription = DiagnosticListener.AllListeners.Subscribe(this);
+            var telemetryClient = new TelemetryClient(configuration);
+            telemetryClient.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("dsl:");
+
+            this.client = telemetryClient;
+
+            // Protect against multiple subscriptions if Initialize is called twice
+            if (this.allDiagnosticListenersSubscription == null)
```

Comment:
should it be locked like here: https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/8645f55e16284a8468c40bc34c2ca7091e92b547/Src/DependencyCollector/Shared/DependencyTrackingTelemetryModule.cs#L121-L126

Reply:
OK. Uses CAS now.